sdk/eventhub/azure-eventhubs-checkpointstoreblob-aio/azure/eventhub/extensions/checkpointstoreblobaio/_blobstoragepmaio.py

```diff
@@ -0,0 +1,208 @@
+# --------------------------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for license information.
+# --------------------------------------------------------------------------------------------
+from typing import Iterable, Dict, Any, Optional
+import logging
+from collections import defaultdict
+import asyncio
+from azure.eventhub import OwnershipLostError  # type: ignore  #pylint:disable=no-name-in-module
+from azure.eventhub.aio import PartitionManager  # type: ignore
+from azure.core.exceptions import ResourceModifiedError, ResourceExistsError  # type: ignore
+from azure.core import MatchConditions
+from azure.storage.blob.aio import ContainerClient, BlobClient  # type: ignore
+
+logger = logging.getLogger(__name__)
+UPLOAD_DATA = ""
+
+
+class BlobCheckpointStore(PartitionManager):
+    """A CheckpointStore that uses Azure Blob Storage to store the partition ownership and checkpoint data.
+
+    This class implements methods list_ownership, claim_ownership, update_checkpoint and list_checkpoints that are
+    defined in class azure.eventhub.aio.CheckpointStore of package azure-eventhub.
+
+    :param str blob_account_url:
+        The URI to the storage account.
+    :param container_name:
+        The name of the container for the blobs.
+    :type container_name: str
+    :param credential:
+        The credentials with which to authenticate. This is optional if the
+        account URL already has a SAS token. The value can be a SAS token string, an account
+        shared access key, or an instance of a TokenCredentials class from azure.identity.
+        If the URL already has a SAS token, specifying an explicit credential will take priority.
+    """
+    def __init__(self, blob_account_url, container_name, credential=None, **kwargs):
```

Comment:
Just to confirm, does the **kwargs have anything that needs to reference Storage Blob library?
For this async code, after argument "credential", the **kwargs should be changed to *, arg1=..., arg2=...?

Reply:
Yup I can use the Py3 "*" syntax here.
Although I'm rather hesistant to include _all_ the possible options for Storage... I'll see if I can filter down a relevant list.

Reply:
I've used the syntax for the `credential` parameter, however looking at the docs I don't think there's any further keyword args we should expose here - they're largely related to blob upload chunking which isn't relevant.